### PR TITLE
add pytorch faster rcnn rpn intermediate objectness extraction

### PR DESCRIPTION
**Pull request by GuluDeemo**
* Add model warm up before measuring the model inference latency. It makes the latency results more accurate.
* Modify the pytorch faster rcnn model scripts to extract the intermediate objectness from region proposal network. (Because the dds code only support faster rcnn resnet101, I chose to modify the pytorch scripts)

First two images are the **detection results** from faster rcnn mobilenetv2
**The accuracy of current fasterrcnn mobilenetv2 is not good enough. The reason I guess is that the example scripts given by pytorch-torchvision didn't construct a FPN network for mobilenetv2. I need to learn about the structure of mobilenetv2 and try FPN network if we need to improve the accuracy in the future** 
I suggest we could use faster rcnn resnet 18 and resnet 101 or 152 as our fog model and cloud model, which will ease a lot.

0-det

62-det

These are the proposal results from faster rcnn mobilenetv2 **region proposal network,** the number is **objectness**.

62-rpn

0-rpn

### CODE REVIEW

**Comment by GuluDeemo** [comment]
Add more clear yolo implementation and extract objectness of yolo:

## **Yolov3-tiny detection results:**
11-det

## **Yolov3-tiny objectness results:**
11-rpn

**Comment by HuaizhengZhang** [comment]
Will check after MLsys

**Comment by HuaizhengZhang** [comment]
agree

No need to waste time on mobinet, resnet18 is great.

File: cfg/yolov3-tiny.cfg [hunk 1]
```diff
@@ -172,11 +172,11 @@ filters=255
 activation=linear
 
 [yolo]
-mask = 0,1,2
+mask = 1,2,3
```
Comment (HuaizhengZhang) [inline cfg/yolov3-tiny.cfg:175]: Why change?

Do we have any backup?

File: cfg/yolov3.cfg [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 # Testing
-batch=1
-subdivisions=1
+#batch=1
+#subdivisions=1
 # Training
-# batch=64
-# subdivisions=16
+batch=16
+subdivisions=1
```
Comment (HuaizhengZhang) [inline cfg/yolov3.cfg:7]: it is better not to change the default values.

we are building the system and we don't need to tune the performance in the begining

**Comment by HuaizhengZhang** [comment]
还有就是感觉这次堆了好多代码啊，具体作用需要弄清楚

**Comment by GuluDeemo** [comment]
我之后弄一个完整的文档，展示核心代码部分，核心代码其实堆得比较少。然后我会把一些冗余代码精简掉。

File: cfg/yolov3-tiny.cfg (hunk 1 above)
Comment (GuluDeemo) [reply cfg/yolov3-tiny.cfg:175]: 我换了一个yolov3的实现，之前的那个是2年前的应该是基于pytorch0.4.0的，然后他的yolov3-tiny的结果也一直有问题。我就换了一个更新的实现repo。并且把代码堆在了detector/yolo_model里面，因为我需要去把里面的一些中间变量取出来，比如objectness。但是代码里面还有很多冗余，应该之后可以精简掉一些用不上的代码块。

我已经把yolov3和fasterrcnn的输出align起来的，他们应该是对齐的。（之前yolov3的输出是tensors，fasterrcnn输出是list(tensors)之类的不对齐，xyxy或者xywh这种对齐的问题）还有一些后处理的问题。

File: cfg/yolov3.cfg (hunk 2 above)
Comment (GuluDeemo) [reply cfg/yolov3.cfg:7]: 应该是换了个repo，然后里面的cfg不一样，这个cfg是更新的repo里面默认的。
